op/framework.py

```diff
@@ -374,14 +372,21 @@ def __init__(self, data_path, charm_dir, meta, model):
         self.charm_dir = charm_dir
         self.meta = meta
         self.model = model
-        self._event_count = 0
         self._observers = []      # [(observer_path, method_name, parent_path, event_key)]
         self._observer = weakref.WeakValueDictionary()       # {observer_path: observer}
         self._type_registry = {}  # {(parent_path, kind): cls}
         self._type_known = set()  # {cls}
 
         self._storage = SQLiteStorage(data_path)
 
+        # We can't use the higher-level StoredState because it relies on events.
+        self.register_type(StoredStateData, None, StoredStateData.handle_kind)
+        self._stored = StoredStateData(self, '_stored')
+        try:
+            self._stored = self.load_snapshot(self._stored.handle)
+        except NoSnapshotError:
+            self._stored['event_count'] = 0
```

Comment:
Also, can we please start events at 1?  This way we can eventually assume that 0 and uninitialized mean the same thing without messing anything up.

Thanks.